### ISSUE TITLE
Runtime Error Stack expects non-empty TensorList line 167

### ISSUE THREAD

**Issue by subhendukhatuya** [shuhao02/RouterDC#5]
In train_router_mdeberta.py line 167 (under compute_cluster_loss function) we are getting empty list (all_index) to do torch.stack operation for some cluster ids and it throws Runtime Error Stack expects non-empty TensorList

**Comment by shuhao02** [comment]
Can you provide more details?

We can not reproduce your error 😂

**Comment by subhendukhatuya** [comment]
This error occurs when cluster  list is passed with all same cluster
ids(let's say all 1). In that case it could not find negative samples. So
torch.stack operation fails. You can run gsm8 with seed as 5, then you will
get runtime error for cluster id list having all 1.
In that case it it fine if we return the loss 0.0?
